authentik/events/models.py

```diff
@@ -196,8 +243,10 @@ def new(
             current = currentframe()
             parent = current.f_back
             app = parent.f_globals["__name__"]
+
         cleaned_kwargs = cleanse_dict(sanitize_dict(kwargs))
-        event = Event(action=action, app=app, context=cleaned_kwargs)
+        batch = EventBatch.get_or_create_batch(action)
```

Comment:
shouldn't create the batch here since the event isn't saved yet, this should be created in the .save() method

Reply:
Done!